packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +36,63 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold geometry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  const ScaffoldGeometry({
+    this.bottomNavigationBarTop,
+    this.floatingActionButton,
+    this.floatingActionButtonScale = 1.0,
+  });
+
+  /// The distance from the scaffold's top edge to the top edge of the
+  /// rectangle in which the [Scaffold.bottomNavigationBar] bar is being laid
+  /// out.
+  ///
+  /// When there is no [Scaffold.bottomNavigationBar] set, this will be null.
+  final double bottomNavigationBarTop;
+
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.floatingActionButton].
+  ///
+  /// This can have any value when there is no [Scaffold.floatingActionButton]
+  /// set.
+  final Rect floatingActionButton;
```

Comment:
is a Rect really the best way to describe this? Consider a FAB whose shape is a downwards-pointing triangle.

I'd probably just go with an offset to the center, and a radius?

Anyway, a Rect is fine if you think it's better than that.

Reply:
Following f2f we keep it a rect, as this keeps the notch computation logic similar to the path computation logic for the ShapeBorder (both take a Rect as their input)